langchain4j-open-ai-official/src/main/java/dev/langchain4j/model/openaiofficial/OpenAiOfficialChatModel.java

```diff
@@ -0,0 +1,491 @@
+                        .toolSpecifications(toolSpecification)
+                        .toolChoice(REQUIRED)
+                        .build())
+                .build();
+        ChatResponse chatResponse = chat(chatRequest);
+        return convertResponse(chatResponse);
+    }
+
+    @Override
+    public ChatResponse doChat(ChatRequest chatRequest) {
+
+        OpenAiOfficialChatRequestParameters parameters = (OpenAiOfficialChatRequestParameters) chatRequest.parameters();
+        InternalOpenAiOfficialHelper.validate(parameters);
+
+        if (this.useAzure) {
+            if (!parameters.modelName().equals(this.azureModelName)) {
+                throw new UnsupportedFeatureException(
+                        "On Azure OpenAI, it is not supported to change the modelName, as it's part of the deployment URL");
+            }
+        }
+
+        ChatCompletionCreateParams chatCompletionCreateParams = toOpenAiChatCompletionCreateParams(
+                        chatRequest, parameters, strictTools, strictJsonSchema)
+                .build();
+
+        // Unlike other LangChain4j modules, this doesn't use the `withRetry` method because the OpenAI SDK already has
+        // retry logic included
+        ChatCompletion chatCompletion = client.chat().completions().create(chatCompletionCreateParams);
+
+        OpenAiOfficialChatResponseMetadata.Builder responseMetadataBuilder =
+                OpenAiOfficialChatResponseMetadata.builder()
+                        .id(chatCompletion.id())
+                        .modelName(chatCompletion.model())
+                        .created(chatCompletion.created());
+
+        if (!chatCompletion.choices().isEmpty()) {
+            responseMetadataBuilder.finishReason(
+                    finishReasonFrom(chatCompletion.choices().get(0).finishReason()));
+        }
+        if (chatCompletion.usage().isPresent()) {
+            responseMetadataBuilder.tokenUsage(
+                    tokenUsageFrom(chatCompletion.usage().get()));
+        }
+        if (chatCompletion.serviceTier().isPresent()) {
+            responseMetadataBuilder.serviceTier(
+                    chatCompletion.serviceTier().get().toString());
+        }
+        if (chatCompletion.systemFingerprint().isPresent()) {
+            responseMetadataBuilder.systemFingerprint(
+                    chatCompletion.systemFingerprint().get());
+        }
+
+        return ChatResponse.builder()
+                .aiMessage(aiMessageFrom(chatCompletion))
+                .metadata(responseMetadataBuilder.build())
+                .build();
+    }
+
+    public static OpenAiOfficialChatModelBuilder builder() {
+        for (OpenAiOfficialChatModelBuilderFactory factory :
+                loadFactories(OpenAiOfficialChatModelBuilderFactory.class)) {
+            return factory.get();
+        }
+        return new OpenAiOfficialChatModelBuilder();
+    }
+
+    public static class OpenAiOfficialChatModelBuilder {
```

Comment:
```suggestion
    public static class Builder {
```

Please rename it (in all classes), as it is quite verbose

Reply:
Done